py-polars/polars/lazyframe/frame.py

```diff
@@ -1674,7 +1677,7 @@ def profile(
         truncate_nodes: int = 0,
         figsize: tuple[int, int] = (18, 8),
         streaming: bool = False,
-        engine: EngineType = "cpu",
+        engine: EngineType = "in-memory",
```

Comment:
Can we set all of these to `"auto"`? Then we can decide what to use. Currently this already the case as the in-memory engine used to go into streaming in certain conditions.